src/ui/docengine.cpp

```diff
@@ -389,6 +392,58 @@ void DocEngine::unmonitorDocument(const QString &fileName)
     }
 }
 
+
+QString DocEngine::getAvailableSudoProgram() const
+{
+    QProcess p;
+
+    p.start("which kdesu");
+    p.waitForFinished(10);
+    if (p.exitCode() == 0) return "kdesu";
+
+    p.start("which gksu");
+    p.waitForFinished(10);
+    if (p.exitCode() == 0) return "gksu";
+
+    return "";
+}
+
+bool DocEngine::trySudoSave(QString sudoProgram, QUrl outFileName, Editor* editor) {
+    if(sudoProgram.isEmpty())
+        return false;
+
+    QString filePath = PersistentCache::createValidCacheName(
+                PersistentCache::cacheDirPath(),
+                outFileName.fileName() )
+            .toLocalFile();
+
+    QFile file(filePath);
+
+    if (!write(&file, editor))
+        return false;
+
+    QProcess p;
+
+    if (sudoProgram == "kdesu")
+        p.start("kdesu", QStringList()
+                << "--noignorebutton"
+                << "-n"
+                << "-c" << "cp" << filePath << outFileName.toLocalFile());
+    else if (sudoProgram == "gksu")
+        p.start("gksu", QStringList()
+                << "-S" << tr("-m Notepadqq asks permission to overwrite the following file:\n\n%1")
```

Comment:
-m should be outside of tr().

Also, it's better if you quote the string, so that translation won't mess up with it. For example, you could use `'` as the quoting character and then replace any occurrence of `'` in the string with `'"'"'`

Reply:
or well, instead of manually quoting, just add a separate entry in the QStringList

Reply:
Changed the `-m` part. 

I don't think quoting the string is needed. Pretty sure that `Process::start()` automatically quotes arguments in the string list that contains whitespace because I initially had the whole string quoted manually and the quotes would actually show in the GUI.